app/controllers/users/reactivate_profile_controller.rb

```diff
@@ -19,7 +19,12 @@ def create
     def build_reactivate_profile_form
       ReactivateProfileForm.new(
         current_user,
-        params[:reactivate_profile_form].permit(:recovery_code, :password)
+        params[:reactivate_profile_form].
+        permit({:recovery_code => []}, :password).
+        tap do |obj|
+          code = obj[:recovery_code]
+          obj[:recovery_code] = code.join(' ') unless obj[:recovery_code].nil?
```

Comment:
It seemed easier to do it here, since there didn't seem a good place to manipulate the recovery code. I guess after the super, maybe?

The second comment here is odd to me, why does the order of the keys matter to rails? For example, if I put `permit(recovery_code: [], :password)` I get an error.